soc/arm/st_stm32/Kconfig

```diff
@@ -14,6 +15,13 @@ config SOC_FAMILY
 	string
 	default "st_stm32"
 
+config STM32_ENABLE_DEBUG_SLEEP_STOP
+	bool "Allow debugger attach in stop/sleep Mode"
+	help
+	  Some STM32 parts need DBGMCU configuration to allow attach in
+	  sleep/stop states. Enable this to allow to attach w/o resetting
+	  the core (useful for debugging).
```

Comment:
Description should be updated as the impact is larger than pure debugger attach in stop/sleep mode